aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -249,16 +254,20 @@ public class MyDep : IMyDep1, IMyDep2 {}
 
 services.TryAddEnumerable(ServiceDescriptor.Singleton<IMyDep1, MyDep>());
 services.TryAddEnumerable(ServiceDescriptor.Singleton<IMyDep2, MyDep>());
-// Two registrations of MyDep for IMyDep1 is avoided by the following line:
 services.TryAddEnumerable(ServiceDescriptor.Singleton<IMyDep1, MyDep>());
 ```
 
+Service registration is generally not order dependent except when registering multiple implementations of an interface.
+
 ### Constructor injection behavior
 
 Services can be resolved by two mechanisms:
 
 * <xref:System.IServiceProvider>
-* <xref:Microsoft.Extensions.DependencyInjection.ActivatorUtilities>: Permits object creation without service registration in the dependency injection container. `ActivatorUtilities` is used with user-facing abstractions, such as Tag Helpers, MVC controllers, and model binders.
+* <xref:Microsoft.Extensions.DependencyInjection.ActivatorUtilities>:
+  * Permits object creation without service registration in the dependency injection container.
+  <!-- Review: what is user-facing abstractions? Frame work features or library features? -->
```

Comment:
It looks like this means framework features.

e.g. A controller is an abstraction provided by the framework that users work with closely. Users don't usually register their controllers with DI, but those controllers can have dependencies that come from DI.